tests/templates/test_subroutines/test_qubitization.py

```diff
@@ -0,0 +1,274 @@
+
+    @staticmethod
+    def circuit(coeffs):
+        H = qml.ops.LinearCombination(
+            coeffs, [qml.Y(0), qml.Y(1) @ qml.Y(2), qml.X(0), qml.X(1) @ qml.X(2)]
+        )
+        qml.Qubitization(H, control=(3, 4))
+        return qml.expval(qml.PauliZ(3) @ qml.PauliZ(4))
+
+    # calculated numerically with finite diff method (h = 1e-4)
+    exp_grad = np.array([0.41177729, -0.21262358, 1.64370464, -0.74256522])
+
+    params = np.array([0.4, 0.5, 0.1, 0.3])
+
+    @pytest.mark.autograd
+    def test_qnode_autograd(self):
+        """Test that the QNode executes with Autograd."""
+
+        dev = qml.device("default.qubit")
+        qnode = qml.QNode(self.circuit, dev, interface="autograd")
+
+        params = qml.numpy.array(self.params, requires_grad=True)
+        res = qml.grad(qnode)(params)
+        assert qml.math.shape(res) == (4,)
+        assert np.allclose(res, self.exp_grad, atol=1e-5)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize(
+        "use_jit , shots",
+        ((False, None), (True, None), (False, 50000)),
+    )  # TODO: (True, 50000) fails because jax.jit on jax.grad does not work with AmplitudeEmbedding
+    def test_qnode_jax(self, shots, use_jit):
+        """ "Test that the QNode executes and is differentiable with JAX. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="jax", diff_method=diff_method)
+        if use_jit:
+            qnode = jax.jit(qnode)
+
+        params = jax.numpy.array(self.params)
+
+        jac_fn = jax.jacobian(qnode)
+        if use_jit:
+            jac_fn = jax.jit(jac_fn)
+
+        jac = jac_fn(params)
+        assert jac.shape == (4,)
+        assert np.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("shots", [None])
+    def test_qnode_torch(self, shots):
+        """ "Test that the QNode executes and is differentiable with Torch. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import torch
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="torch", diff_method=diff_method)
+
+        params = torch.tensor(self.params, requires_grad=True)
+        jac = torch.autograd.functional.jacobian(qnode, params)
+        assert qml.math.shape(jac) == (4,)
+        assert qml.math.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("shots", [None, 50000])
+    @pytest.mark.xfail(reason="tf gradient doesn't seem to be working, returns ()")
```

Comment:
Is this because of a bug in other operators?

Reply:
Ive seen it in a few operators. Also, I don't think we are making an effort to support that in the broader PennyLane, so should be fine. Maybe we can confirm with @trbromley

Reply:
Indeed, this is not a particular error of this PR, it is a test that I have seen that is always put with the xfail